File: src/Juniper.Core/HTTP/MediaType/Message.cs
namespace Juniper.HTTP
{
    public partial class MediaType
    {
        public sealed class Message : MediaType
        {
            public Message(string value, string[] extensions = null) : base("message/" + value, extensions) {}

            public static readonly Message CPIM = new Message("cpim");
            public static readonly Message Delivery_Status = new Message("delivery-status");
            public static readonly Message Disposition_Notification = new Message("disposition-notification");
            public static readonly Message Example = new Message("example");
            public static readonly Message External_Body = new Message("external-body");
            public static readonly Message Feedback_Report = new Message("feedback-report");
            public static readonly Message Global = new Message("global");
            public static readonly Message Global_Delivery_Status = new Message("global-delivery-status");
            public static readonly Message Global_Disposition_Notification = new Message("global-disposition-notification");
            public static readonly Message Global_Headers = new Message("global-headers");
            public static readonly Message Http = new Message("http");
            public static readonly Message ImdnXml = new Message("imdn+xml", new string[] {"xml"});

            [System.Obsolete("OBSOLETED by RFC5537")]
            public static readonly Message News = new Message("news");

            public static readonly Message Partial = new Message("partial");
            public static readonly Message Rfc822 = new Message("rfc822", new string[] {"eml", "mime"});
            public static readonly Message S_Http = new Message("s-http");
            public static readonly Message Sip = new Message("sip");
            public static readonly Message Sipfrag = new Message("sipfrag");
            public static readonly Message Tracking_Status = new Message("tracking-status");

            [System.Obsolete("OBSOLETED by request")]
            public static readonly Message VendorSiSimp = new Message("vnd.si.simp");

            public static readonly Message VendorWfaWsc = new Message("vnd.wfa.wsc");

            public static readonly new Message[] Values = {
                CPIM,
                Delivery_Status,
                Disposition_Notification,
                Example,
                External_Body,
                Feedback_Report,
                Global,
                Global_Delivery_Status,
                Global_Disposition_Notification,
                Global_Headers,
                Http,
                ImdnXml,
                Partial,
                Rfc822,
                S_Http,
                Sip,
                Sipfrag,
                Tracking_Status,
                VendorWfaWsc,
            };
        }
    }
}

File: src/Juniper.TypeScript/mediatypes/message.ts
import { deprecate, specialize } from "./util";

const message = specialize("message");

export const Message_CPIM = message("cpim");
export const Message_Delivery_Status = message("delivery-status");
export const Message_Disposition_Notification = message("disposition-notification");
export const Message_Example = message("example");
export const Message_External_Body = message("external-body");
export const Message_Feedback_Report = message("feedback-report");
export const Message_Global = message("global");
export const Message_Global_Delivery_Status = message("global-delivery-status");
export const Message_Global_Disposition_Notification = message("global-disposition-notification");
export const Message_Global_Headers = message("global-headers");
export const Message_Http = message("http");
export const Message_ImdnXml = message("imdn+xml", "xml");
export const Message_News = deprecate(message("news"), "by RFC5537");
export const Message_Partial = message("partial");
export const Message_Rfc822 = message("rfc822", "eml", "mime");
export const Message_S_Http = message("s-http");
export const Message_Sip = message("sip");
export const Message_Sipfrag = message("sipfrag");
export const Message_Tracking_Status = message("tracking-status");
export const Message_Vendor_SiSimp = deprecate(message("vnd.si.simp"), "by request");
export const Message_Vendor_WfaWsc = message("vnd.wfa.wsc");